Example/Tests/BONHumanReadableStringTestCase.m

```diff
@@ -0,0 +1,208 @@
+    BONChain *chain3x = BONChain.new.image(image3x);
+
+    BONAssertEquivalentStrings(chain3x.attributedString, @"{image750x441}");
+}
+
+- (void)testFractionalImagesOfDifferentScales
+{
+    UIImage *fractionalImage2x = [self.class dummyImageOfSize:CGSizeMake(640.5, 376.5) scale:2.0];
+    BONChain *fractionalChain2x = BONChain.new.image(fractionalImage2x);
+
+    BONAssertEquivalentStrings(fractionalChain2x.attributedString, @"{image640.5x376.5}");
+
+    UIImage *fractionalImage3x = [self.class dummyImageOfSize:CGSizeMake(750.0 + kBONOneThird, 441.0 + kBONTwoThirds) scale:3.0];
+    BONChain *fractionalChain3x = BONChain.new.image(fractionalImage3x);
+
+    BONAssertEquivalentStrings(fractionalChain3x.attributedString, @"{image750.333x441.667}");
+}
+
+- (void)testSpecialCharacters
+{
+    BONChain *chain = BONChain.new.string(@"This");
+    [chain appendLink:BONChain.new.string(@"string") separator:BONSpecial.tab];
+    [chain appendLink:BONChain.new.string(@"is") separator:BONSpecial.lineFeed];
+    [chain appendLink:BONChain.new.string(@"populated") separator:BONSpecial.space];
+    [chain appendLink:BONChain.new.string(@"by") separator:BONSpecial.noBreakSpace];
+    [chain appendLink:BONChain.new.string(@"BONSpecial") separator:BONSpecial.enSpace];
+    [chain appendLink:BONChain.new.string(@"characters") separator:BONSpecial.figureSpace];
+    [chain appendLink:BONChain.new.string(@"that") separator:BONSpecial.thinSpace];
+    [chain appendLink:BONChain.new.string(@"will") separator:BONSpecial.hairSpace];
+    [chain appendLink:BONChain.new.string(@"be") separator:BONSpecial.zeroWidthSpace];
+    [chain appendLink:BONChain.new.string(@"replaced") separator:BONSpecial.nonBreakingHyphen];
+    [chain appendLink:BONChain.new.string(@"by") separator:BONSpecial.figureDash];
+    [chain appendLink:BONChain.new.string(@"simple") separator:BONSpecial.enDash];
+    [chain appendLink:BONChain.new.string(@"human") separator:BONSpecial.emDash];
+    [chain appendLink:BONChain.new.string(@"readable") separator:BONSpecial.horizontalEllipsis];
+    [chain appendLink:BONChain.new.string(@"versions") separator:BONSpecial.lineSeparator];
+    [chain appendLink:BONChain.new.string(@"of") separator:BONSpecial.paragraphSeparator];
+    [chain appendLink:BONChain.new.string(@"their") separator:BONSpecial.narrowNoBreakSpace];
+    [chain appendLink:BONChain.new.string(@"own") separator:BONSpecial.wordJoiner];
+    [chain appendLink:BONChain.new.string(@"name") separator:BONSpecial.minusSign];
+
+    NSString *expectedHumanReadableString = @"This{tab}string{lineFeed}is populated{noBreakSpace}by{enSpace}BONSpecial{figureSpace}characters{thinSpace}that{hairSpace}will{zeroWidthSpace}be{nonBreakingHyphen}replaced{figureDash}by{enDash}simple{emDash}human{horizontalEllipsis}readable{lineSeparator}versions{paragraphSeparator}of{narrowNoBreakSpace}their{wordJoiner}own{minusSign}name";
```

Comment:
@adamhrz and this is what happens with some other special characters. Wanted to get any feedback you might have before I merge, since changing this stuff after releasing it would constitute a breaking change.